rust/types/src/metadata.rs

```diff
@@ -315,42 +328,40 @@ Metadata queries
 
 #[derive(Clone, Debug, PartialEq)]
 pub enum Where {
-    DirectWhereComparison(DirectComparison),
+    DirectWhereComparison(DirectWhereComparison),
+    DirectWhereDocumentComparison(DirectDocumentComparison),
```

Comment:
nit: This is weird to me as "where_document/contains" and "where" (for metadata) are distinct in the higher level interface. This leads to a mismatch between how the higher level code creates things and also leads to issues where input types are not "correct by construction". I.e 

https://github.com/chroma-core/chroma/pull/2847/files#diff-32fc56d7903ac5e6cf1f05c11d9b740b1ff09da6977950c74d667caf85b9448fR67

Here a Where clause is allowed for where_document, BUT I could have passed you a DirectWhereComparison instead of a DirectWhereDocumentComparison. Which is incorrect. I'd prefer if we made our code rely on types to validate this sort of thing - in the spirit of "correct by construction".

Reply:
For now I simply remove the `where_document_clause` from the input of `MetadataFilteringOperator` and `MetadataQueryOrchestrator`.  I simply merge `where_clause` and `where_document_clause` as a single clause for the `MetadataQueryOrchestrator`.

Reply:
I agree. This seems like a unification wherein it is easy for upstream code to construct an incorrect predicate and not get caught anywhere. For e.g if the user supplied a `contains <query>, metadata_key <equals> metadata_value`, and the upstream passes `contains <query> OR metadata_key <equals> metadata_value` then it will be accepted in this system. But our explicit contract is always an `AND` between the `where_metadata` and `where_document` clauses

Reply:
Update: we had an offline discussion on this. In the future we would like to support mixed filter on metadata and documents (for example, an `$or` between `{<k>: <v>}` on metadata and `{$contains: <text>}` on document), where it makes sense to unify the ASTs for `Where` and `WhereDocument`. It is fine to keep the implementation like this, but more documentation is needed to explain the context here.